gcc-arm-embedded-toolchain.cmake

```diff
@@ -84,14 +85,27 @@ set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 
 # Create mandatory flags for processor, toolchain etc.
 set(TOOLCHAIN_COMMON_FLAGS "-ffunction-sections -fdata-sections")
+
+set(CORTEX_M0_FLAGS -mcpu=cortex-m0 -mthumb -mabi=aapcs -mfloat-abi=soft)
+set(CORTEX_M3_FLAGS -mcpu=cortex-m3 -mthumb -mabi=aapcs -mfloat-abi=soft)
+set(CORTEX_M4_FLAGS -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfloat-abi=hard -mfpu=fpv4-sp-d16 -fno-strict-aliasing -fno-builtin --short-enums)
+
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL "cortex-m3")
+  string (REPLACE ";" " " FLAGS "${CORTEX_M3_FLAGS}")
   set(TOOLCHAIN_COMMON_FLAGS
-    "${TOOLCHAIN_COMMON_FLAGS} -mcpu=cortex-m3 -mthumb -mabi=aapcs -mfloat-abi=soft"
+    "${TOOLCHAIN_COMMON_FLAGS} ${FLAGS}"
   )
 elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "cortex-m0")
+  string (REPLACE ";" " " FLAGS "${CORTEX_M0_FLAGS}")
   set(TOOLCHAIN_COMMON_FLAGS
-    "${TOOLCHAIN_COMMON_FLAGS} -mcpu=cortex-m0 -mthumb -mabi=aapcs -mfloat-abi=soft"
+    "${TOOLCHAIN_COMMON_FLAGS} ${FLAGS}"
   )
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "cortex-m4")
+  string (REPLACE ";" " " FLAGS "${CORTEX_M4_FLAGS}")
+  set(TOOLCHAIN_COMMON_FLAGS
+    "${TOOLCHAIN_COMMON_FLAGS} $${FLAGS}>"
+  )
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "multiple")
```

Comment:
We were talking to make CMAKE_SYSTEM_PROCESSOR not present for multiple. Also, we shouldn't use CMAKE_ prefix for that (use CMTOOLS instead).